GRVDEV-105 Create Elastic Beanstalk module

TysonBeck opened the pull request.

TysonBeck committed:
GRVDEV-105 Initial commit of eb

jeremyodle commented on aws/elastic-beanstalk/elastic-beanstalk.tf:
```diff
@@ -0,0 +1,594 @@
+
+variable "aliases" {
+  type        = list(string)
+  description = "List of FQDN's - Used to set the Alternate Domain Names (CNAMEs) setting on Cloudfront"
```
What does CloudFront have to do with this? Does EB leverage it?

TysonBeck replied:
Removed as was not needed

jeremyodle commented on aws/elastic-beanstalk/elastic-beanstalk.tf:
```diff
@@ -0,0 +1,594 @@
+# ----------------------------------------------------------------------------------------------------------------------
+
+variable "solution_stack_name" {
+  default = "64bit Amazon Linux 2018.03 v2.12.14 running Docker 18.06.1-ce"
```
Please add descriptions to all variables.

TysonBeck replied:
Added descriptions

jeremyodle commented on aws/elastic-beanstalk/elastic-beanstalk.tf:
```diff
@@ -0,0 +1,594 @@
+# ----------------------------------------------------------------------------------------------------------------------
```
Change base folder of the module to `aws/eb` and this files name to `eb.tf`

jeremyodle commented on aws/elastic-beanstalk/elastic-beanstalk.tf:
```diff
@@ -0,0 +1,594 @@
+# MODULES / RESOURCES
+# ----------------------------------------------------------------------------------------------------------------------
+
+resource "aws_iam_role" "ec2" {
```
When naming resources, put the name of the file in the resources (e.g. `eb_ec2` from `eb.tf`). This is helpful in case you leverage the same resource across different areas of the module.

jeremyodle commented on aws/elastic-beanstalk/elastic-beanstalk.tf:
```diff
@@ -0,0 +1,594 @@
+    from_port   = var.default_process_port
+    to_port     = var.default_process_port
+    protocol    = "6"
+    cidr_blocks = ["0.0.0.0/0"]
```
We should not allow `0.0.0.0/0` if the ALB is internal.

jeremyodle commented on aws/elastic-beanstalk/module.tf:
```diff
@@ -0,0 +1,153 @@
+
+variable "name" {
+  type        = string
+  default     = "elastic-beanstalk"
```
Set default name to `eb`

jeremyodle commented on aws/elastic-beanstalk/module.tf:
```diff
@@ -0,0 +1,153 @@
+
+variable terraform_module {
+  type        = string
+  default     = "gravicore/terraform-gravicore-modules/aws/elastic-beanstalk"
```
`gravicore/terraform-gravicore-modules/aws/eb`

jeremyodle reviewed (changes requested):
I need to review in the console to see if security groups and targets are working, but the following comments should be reviewed.

TysonBeck committed:
Merge branch 'master' into GRVDEV-105-Create-Elastic-Beanstalk-module

TysonBeck committed:
GRVDEV-105 Changed module name

Added ELB scheme to allow LB to be internal or public
Added descriptions
Changed LB SG ingress & egress CIDR to be variablized

TysonBeck committed:
GRVDEV-105 Added DNS record resource

TysonBeck committed:
GRVDEV-105 Corrected the DNS target and added output

TysonBeck committed:
Merge remote-tracking branch 'origin/master' into GRVDEV-105-Create-Elastic-Beanstalk-module
